client/gutenberg/extensions/shared/frontend-management.js

```diff
@@ -28,7 +28,7 @@ export class FrontendManagement {
 		const data = {};
 		for ( const name in attributes ) {
 			const attribute = attributes[ name ];
-			data[ name ] = dataset[ name ];
+			data[ name ] = dataset[ snakeCase( name ) ];
```

Comment:
You might not need this if you use data attributes like `data-map-style` and not `data-map_style`. You should be able to access `node.dataset` with camel-cased keys:

https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset#Name_conversion

Reply:
Data attributes are converted to lowercase, so if I used `data-myAttribute` it becomes `data-myattribute`. My initial approach was to simply lowercase the name `data[ name ] = dataset[ name.toLowerCase() ];` but I switched to the snake_case conversion approach just to be very explicit about what was going on.